values.yaml

```diff
@@ -387,3 +387,116 @@ connectInject:
     # configured proxy.
     proxyDefaults: |
       {}
+
+# Mesh Gateways enable Consul Connect to work across Consul datacenters.
+meshGateway:
+  # If mesh gateways are enabled, a Deployment will be created that runs
+  # gateways and Consul Connect will be configured to use gateways.
+  # See {todo doc link} for more information.
+  # Requirements: consul >= 1.6.0 and consul-k8s >= 0.8.2 if using global.bootstrapACLs.
+  enabled: false
+
+  # Which mode the gateway should run in. Can be either 'remote' or 'local'.
+  # If set to remote, Connect traffic to services in other datacenters will
+  # be sent directly to the other datacenter's gateway.
+  # If set to local, Connect traffic to services in other datacenters will
+  # be proxied via the local gateway. Use 'local' if services cannot directly
+  # connect to the other datacenter's gateway and so must proxy through the local
+  # gateway, i.e. if only the gateways can communicate between datacenters.
+  mode: remote
+
+  # Number of replicas for the Deployment.
+  replicas: 2
+
+  # What gets registered as wan address for the gateway.
+  wanAddress:
+    # Port that gets registered.
+    port: 443
+
+    # If true, each Gateway Pod will advertise its NodeIP
+    # (as provided by the Kubernetes downward API) as the wan address.
+    # This is useful if the node IPs are routable from other DCs.
+    # The port key will be used as the port for the wan address.
+    useNodeIP: true
+
+    # If true, each Gateway Pod will advertise its NodeName
+    # (as provided by the Kubernetes downward API) as the wan address.
+    # Again, this is useful if the node names are DNS entries that are
+    # routable from other DCs.
+    # The port key will be used as the port for the wan address.
+    # Must set useNodeIP to false.
+    # useNodeName: false
+
+    # If set, this will be the host of the wan address for all gateways.
+    # Users must ensure that this address routes to the Gateway pods,
+    # for example via a DNS
+    # entry that routes to the Service fronting the Deployment.
+    # The port key will be used as the port for the wan address.
+    # Must set useNodeIP and useNodeName to false.
+    # host: myhostname.com
+
+  # The service key configures the Service that fronts the Gateway Deployment.
+  service:
+    # Whether to create a Service or not.
+    enabled: false
+    type: LoadBalancer
+    port: 443
+
+    # Optional YAML string for additional annotations.
+    annotations: |
+    # Optional YAML string that will be appended to the Service spec.
+    additionalSpec: |
+
+  # Envoy image to use.
+  imageEnvoy: envoyproxy/envoy:v1.8.0
+
+  # If set to true, gateway Pods will run on the host network.
+  hostNetwork: false
+
+  # dnsPolicy to use.
+  dnsPolicy: ClusterFirst
+
+  # Override the default 'mesh-gateway' service name registered in Consul.
+  # Cannot be used if bootstrapACLs is true since the ACL token generated
```

Comment:
There should definitely be a test for this, since it would be good to have a hard requirement on this in the templating.

Reply:
How can I do a hard requirement?

Reply:
Where the `consulServiceName` is used and `bootstrapACLs` is true, you can require that the service name is equal to `mesh-gateway` or that section is not rendered.

Reply:
I'm going to do this via a `fail` call. I'll tag you on it, lmk what you think.